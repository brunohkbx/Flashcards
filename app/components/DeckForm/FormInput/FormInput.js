import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { HelperText, TextInput } from 'react-native-paper';
import { get } from 'lodash/object';

const FormInput = (
  { field: { name, value }, form: { touched, errors }, ...props}
  ) => {
  const error = !!(get(touched, name) && get(errors, name));

  return (
    <View>
      <TextInput
        error={error}
        value={value}
        {...props}
      />
      {error &&
        <HelperText type="error" visible={error}>
          {get(errors, name)}
        </HelperText>
      }
    </View>
  );
};

FormInput.propTypes = {
  field: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired
};

export default FormInput;
