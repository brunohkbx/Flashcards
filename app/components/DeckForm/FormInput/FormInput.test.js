import React from 'react';
import { shallow } from 'enzyme';
import FormInput from './FormInput';

describe('FormInput', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      field: { name: 'title', value: 'React' },
      form: { touched: {}, errors: {} }
    }, propOverrides);

    const wrapper = shallow(<FormInput {...defaultProps} />);

    return {
      wrapper
    };
  };

  it('renders properly', () => {
    const { wrapper } =  setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('displays an error message when the input is invalid', () => {
    const { wrapper } = setup({
      form: {
        touched: { title: true },
        errors: { title: 'Required' }
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});
