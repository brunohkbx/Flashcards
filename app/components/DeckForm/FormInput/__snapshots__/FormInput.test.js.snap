// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput displays an error message when the input is invalid 1`] = `
<Component>
  <withTheme(TextInput)
    error={true}
    value="React"
  />
  <withTheme(HelperText)
    type="error"
    visible={true}
  >
    Required
  </withTheme(HelperText)>
</Component>
`;

exports[`FormInput renders properly 1`] = `
<Component>
  <withTheme(TextInput)
    error={false}
    value="React"
  />
</Component>
`;
